src/test/kotlin/de/bund/digitalservice/useid/refresh/RefreshControllerIntegrationTest.kt

```diff
@@ -20,8 +20,7 @@ import java.util.UUID
 private const val AUTHORIZATION_HEADER = "Bearer some-api-key"
 
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
-@Tag("integration")
-class RefreshControllerIntegrationTest(@Autowired val webTestClient: WebTestClient) {
+class RefreshControllerIntegrationTest(@Autowired val webTestClient: WebTestClient) : PostgresTestcontainerIntegrationTest() {
```

Comment:
why does `RefreshControllerIntegrationTest` need to extend `PostgresTestcontainerIntegrationTest` ? is it because of the Controller uses repository?

Reply:
The refresh endpoint uses the `IdentificationSessionService` which relies on the `IdentificationSessionRepository`. Furthermore, the test creates an identification session and fetches a tc token which are both operations that require the mentioned service and repository to work. Hence, either we need a test databse (e.g. by extending `PostgresTestcontainerIntegrationTest`) or we need to mock the service / repository.